addons/common/fnc_unitHasSpace.sqf

```diff
@@ -0,0 +1,129 @@
+#include "script_component.hpp"
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_unitHasSpace
+
+Description:
+    Checks if unit has enough free space in inventory to store item.
+
+    Doesn't take current unit load into account unlike canAdd command.
+
+Parameters:
+    _unit          - Unit <OBJECT>
+    _item          - Item to store <STRING>
+    _count         - Item count <NUMBER> (Default: 1)
+    _checkUniform  - Check space in uniform <BOOLEAN> (Default: true)
+    _checkVest     - Check space in vest <BOOLEAN> (Default: true)
+    _checkBackpack - Check space in backpack <BOOLEAN> (Default: true)
+
+Returns:
+    True if unit has free space, false otherwise <BOOLEAN>
+
+Examples:
+    (begin example)
+        [player, "acc_flashlight"] call CBA_fnc_unitHasSpace
+        [player, "30Rnd_556x45_Stanag", 7, false, true, false] call CBA_fnc_unitHasSpace
+    (end)
+
+Author:
+    Dystopian
+---------------------------------------------------------------------------- */
+SCRIPT(unitHasSpace);
+
+params [
+    "_unit",
+    "_item",
+    ["_count", 1],
+    ["_checkUniform", true],
+    ["_checkVest", true],
+    ["_checkBackpack", true]
+];
+
+#define TYPE_VEST 701
+#define TYPE_UNIFORM 801
+#define TYPE_BACKPACK 901
+
+if (isNil QGVAR(itemMassAllowedSlots)) then {
+    LOG("create ns");
+    GVAR(itemMassAllowedSlots) = [] call CBA_fnc_createNamespace;
+};
+
+private ["_mass", "_allowedSlots"];
+private _itemMassAllowedSlots = GVAR(itemMassAllowedSlots) getVariable _item;
+
+if (!isNil "_itemMassAllowedSlots") then {
+    _mass = _itemMassAllowedSlots # 0;
+    _allowedSlots = _itemMassAllowedSlots # 1;
+    TRACE_3("cache found",_item,_mass,_allowedSlots);
+} else {
```

Comment:
```sqf
GVAR(itemMassAllowedSlots) getVariable [_item, []] params ["_mass", "_allowedSlots"];

if (isNil "_mass") then {
```

Reply:
this was my first version but without `, []` part so I got error and had to rewrite it. Thanks :)